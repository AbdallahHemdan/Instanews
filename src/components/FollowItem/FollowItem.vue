<template>
  <div class="follow-item">
    <img :src="following.img" :alt="following.name + 'image'" class="follow-item__icon" />

    <div class="follow-item__right">
      <div class="follow-item__info">
        <div class="follow-item__username">{{ following.name }}</div>
        <div class="follow-item__nickname">
          {{ following.nickName }}
        </div>
      </div>

      <button class="follow-item__cta">Follow</button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'FollowItem',
  props: {
    following: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.follow-item {
  display: flex;
  align-items: center;
  margin-bottom: $font-size-2x;

  &__icon {
    width: 36px;
    height: 36px;
    border-radius: 50%;
  }

  &__right {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-left: 8px;
  }

  &__info {
    margin-left: 0.5rem;
  }

  &__nickname {
    font-size: $font-size-1x;
    color: $dark-gray;
  }

  &__cta {
    border: 1px solid transparent;
    background-color: $main-color;
    border-radius: 4px;
    color: $white;
    font-weight: 600;
    padding: 0 $font-size-3x;
    font-size: $font-size-2x;
    height: 32px;
  }
}
</style>
