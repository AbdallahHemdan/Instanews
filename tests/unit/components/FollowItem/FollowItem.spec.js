import { shallowMount } from '@vue/test-utils';
import FollowItem from './../../../../src/components/FollowItem/FollowItem.vue';

describe('FollowItem.vue', () => {
  let cmp;
  const following = {
    img: require('@/assets/Suggestions/5.jpg'),
    name: 'morgan',
    nickName: 'Morgan Richardson',
  };

  beforeEach(() => {
    cmp = shallowMount(FollowItem, {
      propsData: {
        following,
      },
    });
  });

  describe('Testing props', () => {
    it('should render follow user image src correctly', () => {
      expect(cmp.find('.follow-item__icon').attributes('src')).toBe(following.img);
    });

    it('should render follow user image alt text correctly', () => {
      expect(cmp.find('.follow-item__icon').attributes('alt')).toBe(following.name + 'image');
    });

    it('should render follow user name correctly', () => {
      expect(cmp.find('.follow-item__username').text()).toBe(following.name);
    });

    it('should render follow user nickname correctly', () => {
      expect(cmp.find('.follow-item__nickname').text()).toBe(following.nickName);
    });
  });

  describe('Test inner text of elements', () => {
    it('should render follow cta inner text correctly', () => {
      expect(cmp.find('.follow-item__cta').text()).toBe('Follow');
    });
  });

  describe('Snapshot testing', () => {
    it('should pass snapshot testing', () => {
      expect(cmp.element).toMatchSnapshot();
    });
  });
});
