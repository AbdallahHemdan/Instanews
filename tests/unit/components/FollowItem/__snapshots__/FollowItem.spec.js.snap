// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowItem.vue Snapshot testing should pass snapshot testing 1`] = `
<div
  class="follow-item"
>
  <img
    alt="morganimage"
    class="follow-item__icon"
    src=""
  />
   
  <div
    class="follow-item__right"
  >
    <div
      class="follow-item__info"
    >
      <div
        class="follow-item__username"
      >
        morgan
      </div>
       
      <div
        class="follow-item__nickname"
      >
        
        Morgan Richardson
      
      </div>
    </div>
     
    <button
      class="follow-item__cta"
    >
      Follow
    </button>
  </div>
</div>
`;
